datasets/imppres/imppres.py

```diff
@@ -0,0 +1,318 @@
+_LICENSE = "Creative Commons Attribution-NonCommercial 4.0 International Public License"
+
+# The HuggingFace dataset library don't host the datasets but only point to the original files
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+_URLs = {"default": "https://github.com/facebookresearch/Imppres/blob/master/dataset/IMPPRES.zip?raw=true"}
+
+
+class Imppres(datasets.GeneratorBasedBuilder):
+    """Each sentence type in IMPPRES is generated according to a template that specifies the linear order of the constituents in the sentence. The constituents are sampled from a vocabulary of over 3000 lexical items annotated with grammatical features needed to ensure wellformedness. We semiautomatically generate IMPPRES using a codebase developed by Warstadt et al. (2019a) and significantly expanded for the BLiMP dataset (Warstadt et al., 2019b)."""
+
+    VERSION = datasets.Version("1.1.0")
+
+    # This is an example of a dataset with multiple configurations.
+    # If you don't want/need to define several sub-sets in your dataset,
+    # just remove the BUILDER_CONFIG_CLASS and the BUILDER_CONFIGS attributes.
+
+    # If you need to make complex sub-parts in the datasets with configurable options
+    # You can create your own builder configuration class to store attribute, inheriting from datasets.BuilderConfig
+    # BUILDER_CONFIG_CLASS = MyBuilderConfig
+
+    # You will be able to load one or the other configurations in the following list with
+    # data = datasets.load_dataset('my_dataset', 'first_domain')
+    # data = datasets.load_dataset('my_dataset', 'second_domain')
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="presupposition",
+            version=VERSION,
+            description="Presuppositions are facts that the speaker takes for granted when uttering a sentence.",
+        ),
+        datasets.BuilderConfig(
+            name="implicature",
+            version=VERSION,
+            description="Scalar implicatures are inferences which can be drawn when one member of a memorized lexical scale is uttered.",
+        ),
+    ]
+
+    # DEFAULT_CONFIG_NAME = "first_domain"  # It's not mandatory to have a default configuration. Just use one if it make sense.
+
+    def _info(self):
+        if (
+            "presupposition" in self.config.name
+        ):  # This is the name of the configuration selected in BUILDER_CONFIGS above
+            features = datasets.Features(
+                {
+                    "premise": datasets.Value("string"),
+                    "hypothesis": datasets.Value("string"),
+                    "trigger": datasets.Value("string"),
+                    "trigger1": datasets.Value("string"),
+                    "trigger2": datasets.Value("string"),
+                    "presupposition": datasets.Value("string"),
+                    "gold_label": datasets.Value("string"),
```

Comment:
Can you rename use the ClassLabel feature type here ?

For example
```
"gold_label": datasets.ClassLabel(names=["entailment", "neutral", "contradiction"])
```

Reply:
Done!